bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/handler/ApiBridgeHandler.java

```diff
@@ -0,0 +1,239 @@
+import org.eclipse.jetty.http.HttpHeader;
+import org.eclipse.jetty.http.HttpMethod;
+import org.eclipse.jetty.http.HttpStatus;
+import org.eclipse.jetty.http.HttpStatus.Code;
+import org.openhab.binding.netatmo.internal.api.ApiError;
+import org.openhab.binding.netatmo.internal.api.AuthenticationApi;
+import org.openhab.binding.netatmo.internal.api.NetatmoException;
+import org.openhab.binding.netatmo.internal.api.RestManager;
+import org.openhab.binding.netatmo.internal.api.data.NetatmoConstants.Scope;
+import org.openhab.binding.netatmo.internal.config.ApiHandlerConfiguration;
+import org.openhab.binding.netatmo.internal.config.ApiHandlerConfiguration.Credentials;
+import org.openhab.binding.netatmo.internal.config.BindingConfiguration;
+import org.openhab.binding.netatmo.internal.deserialization.NADeserializer;
+import org.openhab.binding.netatmo.internal.discovery.NetatmoDiscoveryService;
+import org.openhab.binding.netatmo.internal.webhook.NetatmoServlet;
+import org.openhab.core.thing.Bridge;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseBridgeHandler;
+import org.openhab.core.thing.binding.ThingHandlerService;
+import org.openhab.core.types.Command;
+import org.osgi.service.http.HttpService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * {@link ApiBridgeHandler} is the handler for a Netatmo API and connects it to the framework.
+ *
+ * @author Gaël L'hopital - Initial contribution
+ *
+ */
+@NonNullByDefault
+public class ApiBridgeHandler extends BaseBridgeHandler {
+    private static final int TIMEOUT_S = 20;
+
+    private final Logger logger = LoggerFactory.getLogger(ApiBridgeHandler.class);
+    private final BindingConfiguration bindingConf;
+    private @NonNullByDefault({}) ApiHandlerConfiguration thingConf;
+    private final HttpService httpService;
+
+    private Optional<NetatmoServlet> servlet = Optional.empty();
+
+    private final AuthenticationApi connectApi;
+    private final HttpClient httpClient;
+    private final NADeserializer deserializer;
+    private Optional<ScheduledFuture<?>> connectJob = Optional.empty();
+
+    private Map<Class<? extends RestManager>, RestManager> managers = new HashMap<>();
+
+    public ApiBridgeHandler(Bridge bridge, HttpClient httpClient, HttpService httpService, NADeserializer deserializer,
+            BindingConfiguration configuration) {
+        super(bridge);
+        this.bindingConf = configuration;
+        this.httpService = httpService;
+        this.connectApi = new AuthenticationApi(this);
+        this.httpClient = httpClient;
+        this.deserializer = deserializer;
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Initializing Netatmo API bridge handler.");
+        thingConf = getConfigAs(ApiHandlerConfiguration.class);
+        openConnection();
```

Comment:
Blocking network call in `initialize()` method - this should be avoided.

Reply:
Done